### PR TITLE
feat(TDP-2498): react integration appbar

### PR DESCRIPTION
**Pull request by jsomsanith**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-2498

**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)

**(Optional) What is the current behavior?**
(Additional information to the Jira)

**(Optional) What is the new behavior?**
(Additional information to the Jira)
New route `/react` for the new home page

**(Optional) Other information**:

### CODE REVIEW

File: dataprep-webapp/src/app/components/widgets-containers/app-header-bar/app-header-bar.scss [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.nav.navbar-nav {
```
Comment (charlesnguyen) [inline dataprep-webapp/src/app/components/widgets-containers/app-header-bar/app-header-bar.scss]: Add talend licence

File: dataprep-webapp/src/mocks/Settings.mock.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
+const settingsMock = {
```
Comment (charlesnguyen) [inline dataprep-webapp/src/mocks/Settings.mock.js]: Add talend licence?

File: dataprep-webapp/src/app/components/widgets-containers/app-header-bar/app-header-bar-container.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ https://github.com/Talend/data-prep/blob/master/LICENSE
+
+ You should have received a copy of the agreement
+ along with this program; if not, write to Talend SA
+ 9 rue Pages 92150 Suresnes, France
+
+ ============================================================================*/
+
+import AppHeaderBarCtrl from './app-header-bar-controller';
+import './app-header-bar.scss';
```
Comment (charlesnguyen) [inline dataprep-webapp/src/app/components/widgets-containers/app-header-bar/app-header-bar-container.js]: Make sur that the style is designed according to the mockup

File: dataprep-webapp/src/app/settings/settings-service.spec.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+import settings from '../../mocks/Settings.mock';
+
+describe('Settings service', () => {
+	beforeEach(angular.mock.module('app.settings'));
+
+	afterEach(inject((SettingsService) => {
+		SettingsService.clearSettings();
+	}));
+
+	describe('refreshSettings', () => {
```
Comment (charlesnguyen) [inline dataprep-webapp/src/app/settings/settings-service.spec.js]: Need tests for clearSettings?

File: dataprep-webapp/src/app/components/widgets-containers/app-header-bar/app-header-bar.scss (hunk 1 above)
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/widgets-containers/app-header-bar/app-header-bar.scss]: done

File: dataprep-webapp/src/app/components/home/react-home-component.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const HomeComponent = {
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/home/react-home-component.js]: done

File: dataprep-webapp/src/mocks/Settings.mock.js (hunk 2 above)
Comment (jsomsanith) [inline dataprep-webapp/src/mocks/Settings.mock.js]: done

File: dataprep-webapp/src/app/components/widgets-containers/app-header-bar/app-header-bar.scss (hunk 1 above)
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/widgets-containers/app-header-bar/app-header-bar.scss]: done

File: dataprep-webapp/package.json [hunk 6]
```diff
@@ -53,7 +53,10 @@
     "moment-jdateformatparser": "github:zhaozhiming/moment-jdateformatparser",
     "ng-file-upload": "2.2.2",
     "ng-sortable": "^1.3.6",
-    "sunchoke": "git+https://github.com/Talend/sunchoke.git#release/0.1.0-snapshot.5"
+    "ngreact": "^0.3.0",
+    "react": "^15.3.2",
+    "react-talend-components": "^0.7.0",
```
Comment (frassinier) [inline dataprep-webapp/package.json]: `react-bootstrap` is missing, isn't it? And `react-dom` which is a `react-bootstrap` peer dependency too?

File: dataprep-webapp/src/app/components/widgets-containers/app-header-bar/app-header-bar.scss [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+ You should have received a copy of the agreement
+ along with this program; if not, write to Talend SA
+ 9 rue Pages 92150 Suresnes, France
+
+ ============================================================================*/
+
+.nav.navbar-nav {
+	li[role="presentation"] {
+		a[role="button"] {
+			padding-top: 12px;
```
Comment (frassinier) [inline dataprep-webapp/src/app/components/widgets-containers/app-header-bar/app-header-bar.scss]: Ok but why `12px`?

File: dataprep-webapp/package.json (hunk 6 above)
Comment (jsomsanith) [inline dataprep-webapp/package.json]: yes, I removed too many dependencies after previous review :(

File: dataprep-webapp/src/app/components/widgets-containers/app-header-bar/app-header-bar.scss [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+			padding-bottom: 12px;
+		}
+
+		i[class^="icon-"] {
+			&:before {
+				font-size: 1.5em;
```
Comment (jsomsanith) [inline dataprep-webapp/src/app/components/widgets-containers/app-header-bar/app-header-bar.scss]: Changed to `rem`

**Pull request merged**